src/libsystemd/sd-bus/bus-introspect.c

```diff
@@ -150,7 +163,9 @@ int introspect_write_interface(struct introspect *i, const sd_bus_vtable *v) {
 
                 case _SD_BUS_VTABLE_SIGNAL:
                         fprintf(i->f, "  <signal name=\"%s\">\n", v->x.signal.member);
-                        introspect_write_arguments(i, strempty(v->x.signal.signature), NULL);
+                        if (vtable_has_names(vtable))
+                            names = strempty(v->x.method.names);
```

Comment:
indentation... (note that we supply emacs, vim and `.editorconfig` snippets in our tree, to ensure that pretty much any editor you use automatically picks up the right indentation. Hence I am a bit puzzled to see indentation issues like this one: which editor are you suing?)

Reply:
I expected this to be a small modification, so I did it in sublime editor, which works well for other projects that indent with tabs. With spaces, it has problems apparently, it always decides they are 4 and not 8 (I think because broken function declarations are indented a half tab (4 spaces).
I normally use sourceInsight, but on linux it runs on wine and it is not reactive enough for me.